src/assets/CurrencyRateController.ts

```diff
@@ -1,184 +1,216 @@
+    await this.startPolling();
   }
 
   /**
-   * Name of this controller used during composition
+   * Stop polling for the currency rate
    */
-  name = 'CurrencyRateController';
+  stop() {
+    this.stopPolling();
+  }
 
   /**
-   * Creates a CurrencyRateController instance
+   * Prepare to discard this controller.
    *
-   * @param config - Initial options used to configure this controller
-   * @param state - Initial state to set on this controller
+   * This stops any active polling.
    */
-  constructor(
-    config?: Partial<CurrencyRateConfig>,
-    state?: Partial<CurrencyRateState>,
-    fetchExchangeRate = defaultFetchExchangeRate,
-  ) {
-    super(config, state);
-    this.fetchExchangeRate = fetchExchangeRate;
-    this.defaultConfig = {
-      currentCurrency: this.getCurrentCurrencyFromState(state),
-      disabled: true,
-      interval: 180000,
-      nativeCurrency: 'ETH',
-      includeUSDRate: false,
-    };
-    this.defaultState = {
-      conversionDate: 0,
-      conversionRate: 0,
-      currentCurrency: this.defaultConfig.currentCurrency,
-      nativeCurrency: this.defaultConfig.nativeCurrency,
-      usdConversionRate: 0,
-    };
-    this.initialize();
-    this.configure({ disabled: false }, false, false);
-    this.poll();
+  destroy() {
+    super.destroy();
+    this.stopPolling();
   }
 
   /**
    * Sets a currency to track
    *
-   * TODO: Replace this wth a method
-   *
    * @param currentCurrency - ISO 4217 currency code
    */
-  set currentCurrency(currentCurrency: string) {
-    this.activeCurrency = currentCurrency;
-    safelyExecute(() => this.updateExchangeRate());
-  }
-
-  get currentCurrency() {
-    throw new Error('Property only used for setting');
+  async setCurrentCurrency(currentCurrency: string) {
+    this.update((state) => {
+      state.pendingCurrentCurrency = currentCurrency;
+    });
+    await this.updateExchangeRate();
   }
 
   /**
    * Sets a new native currency
    *
-   * TODO: Replace this wth a method
-   *
    * @param symbol - Symbol for the base asset
    */
-  set nativeCurrency(symbol: string) {
-    this.activeNativeCurrency = symbol;
-    safelyExecute(() => this.updateExchangeRate());
+  async setNativeCurrency(symbol: string) {
+    this.update((state) => {
+      state.pendingNativeCurrency = symbol;
+    });
+    await this.updateExchangeRate();
   }
 
-  get nativeCurrency() {
-    throw new Error('Property only used for setting');
+  private stopPolling() {
+    if (this.handle) {
+      clearInterval(this.handle);
+    }
   }
 
   /**
    * Starts a new polling interval
-   *
-   * @param interval - Polling interval used to fetch new exchange rate
    */
-  async poll(interval?: number): Promise<void> {
-    interval && this.configure({ interval }, false, false);
-    this.handle && clearTimeout(this.handle);
+  private async startPolling(): Promise<void> {
+    this.stopPolling();
+    // TODO: Expose polling currency rate update errors
```

Comment:
Clarification: I take it it was never doing this to begin with?

Reply:
Correct